test/li_test.jl

```diff
@@ -0,0 +1,144 @@
+using SpecialFunctions
+using Base.Test
+
+const SF = SpecialFunctions
+
+
+# golden ratio is used in some tests
+ϕ = (sqrt(5)+1)/2.0
+
+@testset "Polylogarithm Li" begin
+    @testset "Check different input formats" begin
+        @test SF.Li(-1.0, 0.0) ≈ 0.0
+        @test SF.Li(-1, 0.0) ≈ 0.0
+        @test SF.Li(-1.0, Complex(0.0)) ≈ 0.0 + 0.0im # don't seem to need imaginary bit here
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0 + 0.0im
+        @test SF.Li(Complex(-1.0), 0.0) ≈ 0.0
+        @test SF.Li(1, 0) ≈ 0.0
+        x = collect(0.0:0.1:0.9)
+        @test all([SF.Li.(1, x)[i] ≈ SF.Li(1, x[i]) for i=1:length(x)])
+        @test SF.Li(Complex(-1.0), Complex(0.3)) ≈ SF.Li(-1.0, 0.3)
```

Comment:
you usually want to test at the boundaries between different approximations (the crossover points of your if statements), very large arguments (near overflow, e.g. 1e300), very small arguments, inf, nan, arguments with very small real or imaginary parts...

this is in addition to random sampling, which is a good fallback test to help catch any corner cases you forgot about.   

When you compare to mpmath, what is the maximum relative error you see in the real or imaginary parts?